src/LocalBackend/LocalLibrary.vala

```diff
@@ -257,36 +257,42 @@ public class Noise.LocalLibrary : Library {
 
     private async void rescan_music_folder_async () {
         var to_remove = new Gee.TreeSet<Media> ();
-        var to_import = new Gee.TreeSet<string> ();
         var files = new Gee.TreeSet<string> ();
 
-        // get a list of the current files
         var music_folder_dir = Settings.Main.get_default ().music_folder;
-        FileUtils.count_music_files (File.new_for_path (music_folder_dir), files);
-        
-        foreach (var m in get_medias ()) {
-            if (!m.isTemporary && !m.isPreview && m.uri.contains (music_folder_dir))
+        var num_items = FileUtils.count_music_files (File.new_for_path (music_folder_dir), files);
+        debug ("Found %d items to import in %s\n", items, music_folder_file);
 
-            if (!File.new_for_uri (m.uri).query_exists ())
-                to_remove.add (m);
-            if (files.contains (m.uri))
-                files.remove (m.uri);
+        foreach (var m in get_medias()) {
+            if (!m.isTemporary && !m.isPreview && m.uri.contains (music_folder_dir)) {
+                if (!File.new_for_uri (m.uri).query_exists ()) {
+                    to_remove.add (m);
+                }
+
+                if (files.contains (m.uri)) {
+                    files.remove (m.uri);
+                }
+            }
```

Comment:
AFAICT that is not what the original code did. The original code was:
```
            if (!m.isTemporary && !m.isPreview && m.uri.contains (music_folder_dir))

            if (!File.new_for_uri (m.uri).query_exists ())
                to_remove.add (m);
            if (files.contains (m.uri))
                files.remove (m.uri);
```

Which is the equivalent of:
```
            if (!m.isTemporary && !m.isPreview && m.uri.contains (music_folder_dir)) {
                if (!File.new_for_uri (m.uri).query_exists ()) {
                    to_remove.add (m);
                }
            }
            if (files.contains (m.uri)) {
                files.remove (m.uri);
             }
```

Is there any reason for changing this?